### PR TITLE
Add ability to hide embedded media

### LINKED ISSUE #1187: Ability to hide embed media
Embed media can catch our attention or pollute chat.

Could be cool to have a way to hide media like in Slack

screen shot 2015-10-21 at 14 55 30

screen shot 2015-10-21 at 14 55 37

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27580374-ability-to-hide-embed-media?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by brakhane**
@RocketChat/core 
Embedded media like images and video can now individually be collapsed

A new preference "Collapse media by default" is also added, the default
is to not collapse media.

Since images are not always shown, "showImage" is also renamed to
"loadImage"

Known limitation: if the same url is used multiple times in the same message, collapse only works for the first one. This is a deficiency in mongodb that already affects the "save bandwidth" feature, so I think it's not a big deal (who posts the same url twice in the same message anyway?)

Closes #1187

### CODE REVIEW

**Comment by jots** [comment]
Does this give the author of the message to hide the preview for all other readers?  I believe that's how it works in slack and I sort of like it.

**Comment by brakhane** [comment]
@jots no, it doesn't.

It's been a while since I've used slack, does Slack give the other the option to hide the media preview by default or the option to not create an "attachment" in the first place (no title, summary, etc)? I think it's the latter.

**Comment by marceloschmidt** [comment]
When I upload an audio and click on collapse, I have the following error:

`collection.js:597 Uncaught MinimongoError: can't set field named $`

image

File: packages/rocketchat-ui/views/app/room.coffee [hunk 1]
```diff
@@ -418,6 +418,14 @@ Template.room.events
 		ChatMessage.update {_id: this._arguments[1]._id, 'urls.url': $(event.currentTarget).data('url')}, {$set: {'urls.$.downloadImages': true}}
 		ChatMessage.update {_id: this._arguments[1]._id, 'attachments.image_url': $(event.currentTarget).data('url')}, {$set: {'attachments.$.downloadImages': true}}
 
+	'click .collapse-switch': (e) ->
+		url = $(e.currentTarget).data('url')
+		collapsed = $(e.currentTarget).data('collapsed')
+		id = @_arguments[1]._id
+		ChatMessage.update {_id: id, 'urls.url': url}, {$set: {'urls.$.collapsed': !collapsed}}
```
Comment (marceloschmidt) [inline packages/rocketchat-ui/views/app/room.coffee]: Check if url is undefined before calling this update (re: https://github.com/RocketChat/Rocket.Chat/pull/2763#issuecomment-205809574)

**Comment by rodrigok** [comment]
I will fix the problem :smile:

**Comment by brakhane** [comment]
@rodrigok Thanks, i was going to fix it this evening, but I'm glad I don't have to.

**Pull request merged**